src/Widgets/FirmwareUpdateRow.vala

```diff
@@ -53,59 +56,107 @@ public class About.Widgets.FirmwareUpdateRow : Gtk.ListBoxRow {
             column_spacing = 12,
             margin = 6
         };
-        grid.attach (icon, 0, 0, 1, 2);
+        grid.attach (image, 0, 0, 1, 2);
         grid.attach (device_name_label, 1, 0);
         grid.attach (version_label, 1, 1);
 
-        switch (device.latest_release.flag) {
-            case Firmware.ReleaseFlag.IS_UPGRADE:
-                if (device.latest_release.version == device.version) {
-                    break;
-                }
+        // TODO: Be smarter to avoid segfault
+        var icons = device.get_icons ();
+        if (icons != null) {
+            for (int i = 0; i < icons.length; i++) {
+                image.icon_name = icons[i];
+            }
```

Comment:
something like this should work:
```suggestion
            image.icon = new GLib.ThemedIcon.from_names (icons);
```

Reply:
It looks like it doesn't, because what we get from `get_icons` isn't necessarily a string array. So probably what needs to be done is to add only values which are strings to a new array and then use themedicon.from_names on that